### PR TITLE
psh: add help command test

### PR DESCRIPTION
**Pull request by damianloew**
## Description

psh: add help command test
## Motivation and Context

JIRA: PD-77

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: psh/test-help.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import psh.tools.psh as psh
+
+HELP_LINE_PATTERN = r'((\s*)(?P<cmd>\w+)(\s+)-(\s+\w+)+(\s*)(\n))'
+
+
+@psh.run
+def harness(p):
+    # asserts that at least 15 help commands are displayed in proper format
+    psh.assert_cmd(p, 'help', rf'{HELP_LINE_PATTERN}{15,}', is_regex=True)
```
Comment (xvuko) [inline psh/test-help.py:23]: {15,} needs to be escaped

File: psh/test.yaml [hunk 2]
```diff
@@ -66,3 +66,6 @@ test:
 
         - name: history
           harness: test-history.py
+
+        - name: help
+          harness: test-history.py
```
Comment (xvuko) [inline psh/test.yaml:71]: test-help

Comment (damianloew) [reply psh/test.yaml]: Thanks, corrected.

**Review by mateusz-bloch — APPROVED** [review]
Well-structured and should serve its purpose efficiently
 
 However, I'm wondering if it's the intended effect that, regardless of the argument given to the `help` function, it's always displayed? If so, I don't see any other test cases, except maybe a test could be added to check, for example, if the function displayed "Unknown command!"  with `HELP` written in uppercase."

**Pull request merged**